File: plugins/OrchidKSS/src/main/java/com/eden/orchid/kss/KssModule.java
package com.eden.orchid.kss;

import com.eden.orchid.OrchidModule;
import com.eden.orchid.api.generators.OrchidGenerator;
import com.eden.orchid.api.resources.resourceSource.PluginResourceSource;
import com.eden.orchid.api.theme.menus.menuItem.OrchidMenuItem;
import com.eden.orchid.kss.menu.StyleguidePagesMenuItemType;
import com.eden.orchid.kss.menu.StyleguideSectionsMenuItemType;

public class KssModule extends OrchidModule {

    @Override
    protected void configure() {
        addToSet(OrchidGenerator.class, KssGenerator.class);

        addToSet(OrchidMenuItem.class,
                StyleguidePagesMenuItemType.class,
                StyleguideSectionsMenuItemType.class);

        addToSet(PluginResourceSource.class,
                KssResourceSource.class);
    }
}

File: plugins/OrchidKSS/src/main/kotlin/com/eden/orchid/kss/KssModule.kt
package com.eden.orchid.kss

import com.eden.orchid.api.generators.OrchidGenerator
import com.eden.orchid.api.registration.OrchidModule
import com.eden.orchid.api.theme.menus.OrchidMenuFactory
import com.eden.orchid.kss.menu.StyleguidePagesMenuItemType
import com.eden.orchid.kss.menu.StyleguideSectionsMenuItemType
import com.eden.orchid.utilities.addToSet

class KssModule : OrchidModule() {

    override fun configure() {
        withResources(20)
        addToSet<OrchidGenerator<*>, KssGenerator>()
        addToSet<OrchidMenuFactory>(
                StyleguidePagesMenuItemType::class,
                StyleguideSectionsMenuItemType::class)
    }
}

